src/hierarchies/lapnig_hierarchy.h

```diff
@@ -0,0 +1,119 @@
+struct State {
+  double mean, scale;
+};
+
+//! Custom container for Hyperparameters values
+struct Hyperparams {
+  double mean, var, shape, scale, mh_log_scale_var, mh_mean_var;
+};
+}  // namespace LapNIG
+
+class LapNIGHierarchy
+    : public BaseHierarchy<LapNIGHierarchy, LapNIG::State, LapNIG::Hyperparams,
+                           bayesmix::LapNIGPrior> {
+ public:
+  LapNIGHierarchy() = default;
+  ~LapNIGHierarchy() = default;
+
+  //! Counters for tracking acceptance rate in MH step
+  static unsigned int accepted_;
+  static unsigned int iter_;
+
+  //! Returns the Protobuf ID associated to this class
+  bayesmix::HierarchyId get_id() const override {
+    return bayesmix::HierarchyId::LapNIG;
+  }
+
+  //! Read and set state values from a given Protobuf message
+  void set_state_from_proto(const google::protobuf::Message &state_) override;
+
+  //! Read and set hyperparameter values from a given Protobuf message
+  void set_hypers_from_proto(
+      const google::protobuf::Message &hypers_) override;
+
+  void save_posterior_hypers() { throw std::runtime_error("Not implemented"); }
+
+  //! Returns whether the hierarchy models multivariate data or not
+  bool is_multivariate() const override { return false; }
+
+  //! Writes current state to a Protobuf message and return a shared_ptr
+  //! New hierarchies have to first modify the field 'oneof val' in the
+  //! AlgoritmState::ClusterState message by adding the appropriate type
+  std::shared_ptr<bayesmix::AlgorithmState::ClusterState> get_state_proto()
+      const override;
+
+  //! Writes current value of hyperparameters to a Protobuf message and
+  //! return a shared_ptr.
+  //! New hierarchies have to first modify the field 'oneof val' in the
+  //! AlgoritmState::HierarchyHypers message by adding the appropriate type
+  std::shared_ptr<bayesmix::AlgorithmState::HierarchyHypers> get_hypers_proto()
+      const override;
+
+  //! Resets summary statistics for this cluster
+  void clear_summary_statistics() override;
+
+  //! Updates hyperparameter values given a vector of cluster states
+  void update_hypers(const std::vector<bayesmix::AlgorithmState::ClusterState>
+                         &states) override;
+
+  //! Updates state values using the given (prior or posterior) hyperparameters
+  LapNIG::State draw(const LapNIG::Hyperparams &params);
+
+  //! Generates new state values from the centering posterior distribution
+  //! @param update_params  Save posterior hypers after the computation?
+  void sample_full_cond(bool update_params = false) override;
+
+ protected:
+  //! Set of values of data points belonging to this cluster
+  std::list<Eigen::RowVectorXd> cluster_data_values;
+
+  //! Samples from the proposal distribution
+  Eigen::VectorXd propose_rwmh(const Eigen::VectorXd &curr_vals);
+
+  //! Evaluates the prior given the mean (unconstrained_parameters(0))
+  //! and log of the scale (unconstrained_parameters(1)) (beware of the
+  //! change of variables!
```

Comment:
This lacks a `)`. Also, what do you mean by the last sentence? You should explain it if it's important